azkaban-web-server/src/main/java/cloudflow/services/ExecutionServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package cloudflow.services;
+
+
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutableFlowBase;
+import azkaban.executor.ExecutableNode;
+import azkaban.executor.ExecutionAttempt;
+import azkaban.executor.ExecutorManagerAdapter;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.executor.Status;
+import cloudflow.error.CloudFlowError;
+import cloudflow.error.CloudFlowException;
+import cloudflow.error.ErrorCode;
+import cloudflow.models.JobExecution;
+import cloudflow.models.JobExecutionAttempt;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExecutionServiceImpl implements ExecutionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExecutionServiceImpl.class);
+    private ExecutorManagerAdapter executorManager;
+
+    @Inject
+    public ExecutionServiceImpl(ExecutorManagerAdapter executorManager) {
+        this.executorManager = executorManager;
+    }
+
+    @Override
+    public Optional<JobExecution> getJobExecution (String executionId, String jobDefinitionId,
+        String user) throws CloudFlowException {
+        // TODO: check user permissions
```

Comment:
Most of these will be addressed in a week or so as soon as the dependencies (projects, flows and jobs) are figured out.
I wouldn't worry too much about the rest since these implementations will likely not be compatible with the API framework of choice.